### ISSUE TITLE
[🐞] Playing Audio on iOS Devices via onClick$ / QRLs does not work

### ISSUE THREAD

**Issue by n8sabes** [BuilderIO/qwik#5064]
### Which component is affected?

Qwik Runtime

### Describe the bug

On iOS devices, auto-playing audio and video+sound in web content is restricted to prevent unexpected data usage. Audio and video playback in iOS Safari requires a direct user action, such as a tap, preventing scripts from auto-playing media without user initiation.

The discovery here is that Apple's iOS security measures ALSO PREVENT playing audio / video+audio from a `QRL` because it's not directly wired to the `.play()` call, requiring **TWO taps**. Even something as simple as deferring the play call within a `setTimeout( () => { audio.play() }, 1000);` will fail to play.

The workaround seems to be to directly wire the `click` event to the play code as shown in the reproduction code, below.

<img width="376" alt="image" src="https://github.com/BuilderIO/qwik/assets/10445528/0aebddcd-6a68-46a9-bc20-b20c900f5d21">

### Reproduction

see attached component code

### Steps to reproduce

Create a component with this code. Then run it on an iOS device.

```tsx
import { component$, useSignal, useStylesScoped$, useVisibleTask$ } from '@builder.io/qwik';

export default component$(() => {
    const audioElementSignal = useSignal<HTMLAudioElement | undefined>();
    const playButtonSignal = useSignal<HTMLButtonElement | undefined>();
    const clickCountSignal = useSignal(0);

    useStylesScoped$(`
    segment {
        width: 100%;
        display: flex;
        flex-direction: column;
        align-items: center;
        text-align: center;
    }    
    button {
        padding: 20px;
        font-weight: bold;
        font-size: 1.2em;
        width: 50%;
        background: green;
        color: white;
    }
    .bad_button {
        background: red;
    }
    `);

    useVisibleTask$(({ track }) => {
        track(() => playButtonSignal.value);
        track(() => audioElementSignal.value);

        const playAudio = () => audioElementSignal.value?.play();

        playButtonSignal.value?.removeEventListener('click', playAudio); // Cleanup
        playButtonSignal.value?.addEventListener('click', playAudio);

        return () => playButtonSignal.value?.removeEventListener('click', playAudio);
    });

    return (
        <segment>
            <audio ref={audioElementSignal} src="https://samplelib.com/lib/preview/mp3/sample-15s.mp3"></audio>
            <h2>iOS Audio Player</h2>
            <h3> [ Run on iOS Device ]</h3>
            <br />
            <h2>Click Count: {clickCountSignal.value}</h2>
            <br />
            <br />
            <button
                class="bad_button"
                onClick$={() => {
                    clickCountSignal.value++;
                    audioElementSignal.value?.play();
                }}
            >
                Play via onClick$ requires
                <br />
                <span style={{ fontWeight: 800, textDecoration: 'underline' }}>TWO</span> taps on iOS devices
            </button>
            <br />
            <br />
            <button ref={playButtonSignal}>
                Play via event listener
                <br />
                works fine
            </button>
            <br />
            <br />
            <button onClick$={() => audioElementSignal.value?.pause()}>Pause</button>
            <br />
        </segment>
    );
});
```

### System Info

```shell
System:
    OS: macOS 13.3.1
    CPU: (8) arm64 Apple M1
    Memory: 77.66 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
  Browsers:
    Chrome: 116.0.5845.96
    Firefox Developer Edition: 117.0
    Safari: 16.4
  npmPackages:
    @builder.io/partytown: ^0.8.0 => 0.8.0
    undici: 5.22.1 => 5.22.1
    vite: 4.4.7 => 4.4.7
```

**Comment by n8sabes** [comment]
Cookbook Recipe:

```tsx
import { component$, useSignal, useStylesScoped$, useVisibleTask$ } from '@builder.io/qwik';

export default component$(() => {
    const audioElementSignal = useSignal<HTMLAudioElement | undefined>();
    const playButtonSignal = useSignal<HTMLButtonElement | undefined>();

    useStylesScoped$(`
    segment {
        width: 100%;
        display: flex;
        flex-direction: column;
        align-items: center;
        text-align: center;
    }    
    button {
        padding: 20px;
        font-weight: bold;
        font-size: 1.2em;
        width: 50%;
        background: green;
        color: white;
    }
    `);

    useVisibleTask$(({ track }) => {
        track(() => playButtonSignal.value);
        track(() => audioElementSignal.value);

        const playAudio = () => audioElementSignal.value?.play();

        playButtonSignal.value?.removeEventListener('click', playAudio);
        playButtonSignal.value?.addEventListener('click', playAudio);

        return () => playButtonSignal.value?.removeEventListener('click', playAudio);
    });

    return (
        <segment>
            <audio ref={audioElementSignal} src="https://samplelib.com/lib/preview/mp3/sample-15s.mp3"></audio>
            <h2>iOS Audio Player</h2>
            <br />
            <br />
            <button ref={playButtonSignal}>Play</button>
            <br />
            <button onClick$={() => audioElementSignal.value?.pause()}>Pause</button>
            <br />
        </segment>
    );
});
```

**Comment by n8sabes** [comment]
Wrote cookbook recipe to explain how to control audio and video media on iOS devices synchronously.

**Comment by gioboa** [comment]
@n8sabes  thanks for sharing. I think this is a good way to solve this clash with play restrictions. Btw Qwik is based on top of QRLs so we can't change this. Can you create a cookbook section for that pls? The section could be generic: "Play audio with Qwik" and you can add this iOS caveat. Thanks in advance

**Comment by n8sabes** [comment]
Closed with #5078.
